tests/js/server/aql/aql-graph-k-shortest-path-0-weight-regression.js

```diff
@@ -0,0 +1,148 @@
+/// @brief tests for query language, graph functions
+///
+/// @file
+///
+/// DISCLAIMER
+///
+/// Copyright 2010-2012 triagens GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Markus Pfeiffer
+/// @author Copyright 2020, ArangoDB GmbH, Cologne, Germany
+////////////////////////////////////////////////////////////////////////////////
+
+const jsunity = require("jsunity");
+const db = require("@arangodb").db;
+const gm = require("@arangodb/general-graph");
+const _ = require("underscore");
+
+const graphName = "UnitTestGraph";
+const vName = "UnitTestVertices";
+const eName = "UnitTestEdges";
+
+const tearDownAll = () => {
+  try {
+    gm._drop(graphName, true);
+  } catch (e) {
+    // Don't care for error, we might runinitially with no graph exist
+  }
+  db._drop(eName);
+  db._drop(vName);
+};
+
+const createGraph = () => {
+  gm._create(graphName, [gm._relation(eName, vName, vName)
+  ], [], {});
+
+  const vertices = [];
+  const edges = [];
+
+  for (var i = 0; i < 19; i++) {
+    vertices.push({
+      _key: i.toString()
+    });
+  }
+
+  edges.push({ _from: `${vName}/0`, _to: `${vName}/1`, weight: 0 });
+  edges.push({ _from: `${vName}/1`, _to: `${vName}/2`, weight: 1 });
+  edges.push({ _from: `${vName}/2`, _to: `${vName}/3`, weight: 0 });
+
+  edges.push({ _from: `${vName}/4`, _to: `${vName}/5`, weight: 0.5});
+  edges.push({ _from: `${vName}/5`, _to: `${vName}/6`, weight: 0});
+  edges.push({ _from: `${vName}/6`, _to: `${vName}/7`, weight: 2.5});
+
+  edges.push({ _from: `${vName}/8`, _to: `${vName}/9`, weight: 0.5});
+  edges.push({ _from: `${vName}/9`, _to: `${vName}/10`, weight: 0});
+  edges.push({ _from: `${vName}/10`, _to: `${vName}/11`, weight: 2.5});
+  edges.push({ _from: `${vName}/11`, _to: `${vName}/12`, weight: 0});
+  edges.push({ _from: `${vName}/12`, _to: `${vName}/13`, weight: 1.0});
+  edges.push({ _from: `${vName}/13`, _to: `${vName}/14`, weight: 2});
+
+  edges.push({ _from: `${vName}/9`, _to: `${vName}/15`, weight: 0});
+  edges.push({ _from: `${vName}/15`, _to: `${vName}/16`, weight: 1});
+  edges.push({ _from: `${vName}/16`, _to: `${vName}/11`, weight: 1});
+
+  edges.push({ _from: `${vName}/9`, _to: `${vName}/17`, weight: 0});
+  edges.push({ _from: `${vName}/17`, _to: `${vName}/18`, weight: 0});
+  edges.push({ _from: `${vName}/18`, _to: `${vName}/14`, weight: 1});
```

Comment:
I am confused because this test does not have negative weights but is a regression test for a bug described as `k shortest paths behaviour wrong with negative weights`?

Reply:
I think this is a typo; I must have writetn "non-negative" somewhere before and the "non" got lost.

Writing "zero weights" is better, anyway.

Reply:
good comment, has been fixed